segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataCleaner.java

```diff
@@ -100,6 +101,24 @@ protected void doStop() {
 
     //endregion
 
+    /**
+     * Persists the metadata of all active Segments from the Container's metadata into the {@link MetadataStore}.
+     * This method does not evict or otherwise perform any cleanup tasks on the Container or its Metadata, nor does it
+     * interfere with the regular operation of {@link #runOnce()}.
+     *
+     * @param timeout Timeout for the operation.
+     * @return A CompletableFuture that, when completed, indicates that the operation completed.
+     */
+    CompletableFuture<Void> persistAll(Duration timeout) {
+        val tasks = this.metadata.getAllStreamSegmentIds().stream()
+                .map(this.metadata::getStreamSegmentMetadata)
+                .filter(Objects::nonNull)
+                .filter(sm -> !sm.isDeleted() && !sm.isMerged())
```

Comment:
Not sure why we don't want to flush deleted and merged segment.

Reply:
The previous step in `LogFlusher` (before this is invoked) tells the `StorageWriter` to apply all the outstanding changes to Storage. That includes any deletions or mergers. Those segments' metadata may still linger in memory (not evicted yet) and should be treated as garbage. No need to write them down here. 

If you look in the `runOnceInternal` method in this class (the one that does evictions), you'll see similar logic.